### PR TITLE
RFR: Edit quiz & WIP quiz

### LINKED ISSUE #164: Edit quiz/book and add WIP property quiz
Books and quizzes need to be able to be changed. Also, quizzes should have the notion of a WIP/published.

### PR DESCRIPTION
**Pull request by micleo2**
:rocket: Ready

Related issues: closes #164

### CODE REVIEW

File: backend/api/models/Quiz.py [hunk 1]
```diff
@@ -11,8 +11,13 @@ class Quiz(Mixin, db.Model):
 
     id = db.Column(db.Integer, unique=True, primary_key=True)
     name = db.Column(db.String, nullable=False)
-    questions = db.relationship("Question", backref="quiz", lazy=True)
+    questions = db.relationship(
```
Comment (Davidw1339) [inline backend/api/models/Quiz.py]: good stuff deleting all related data :+1:

File: backend/api/views/quiz.py [hunk 2]
```diff
@@ -52,27 +52,20 @@ def duplicate_quiz(user_data):
...
 
 
-@quiz.route("/quiz", methods=["POST"])
-def create_quiz():
-    user_data = request.get_json()
-
+def create_quiz_helper(user_data):
     if invalid_quiz_data(user_data):
-        return create_response(
+        return dict(
```
Comment (Davidw1339) [inline backend/api/views/quiz.py]: clever job switching out create_response with dict to reuse create_quiz

File: backend/api/views/quiz.py [hunk 3]
```diff
@@ -271,23 +272,51 @@ def create_quiz_result():
...
+    return create_response(
+        message="Successfully created quiz result; new badges earned",
+        status=200,
+        data={"status": "success"},
+    )
+
+
+@quiz.route("/edit_quiz", methods=["POST"])
+def edit_quiz():
+    pdb.set_trace()
```
Comment (Davidw1339) [inline backend/api/views/quiz.py]: nit: remove this line

File: backend/api/views/quiz.py [hunk 4]
```diff
@@ -271,23 +272,51 @@ def create_quiz_result():
...
+            status=422,
+            data={"status": "fail"},
         )
 
-    db.session.delete(quiz_to_delete)
-    db.session.commit()
+    res = create_quiz_helper(request.get_json())
     return create_response(
-        message="Successfully deleted quiz", status=200, data={"status": "success"}
+        message=res["message"], status=res["status"], data=res["data"]
```
Comment (Davidw1339) [inline backend/api/views/quiz.py]: Might want to switch message from successfully created quiz to successfully edited quiz here

File: backend/api/views/book.py [hunk 5]
```diff
@@ -184,3 +184,30 @@ def delete_quiz(user_id):
...
+            message="Missing required book info", status=422, data={"status": "fail"}
+        )
+
+    book_to_edit = Book.query.get(user_data["book_id"])
+    if book_to_edit is None:
+        return create_response(
+            message="Book not found", status=422, data={"status": "fail"}
+        )
+
+    book_to_edit.name = (user_data["name"],)
```
Comment (Davidw1339) [inline backend/api/views/book.py]: interesting syntax just curious why is do we wrap the name in a tuple?

Comment (micleo2) [reply backend/api/views/book.py]: this was an accident

**Pull request merged**